cluster-autoscaler/simulator/framework/delegating_shared_lister.go

```diff
@@ -118,57 +128,65 @@ func (lister *unsetStorageInfoLister) IsPVCUsedByPods(key string) bool {
 	return false
 }
 
-func (u unsetResourceClaimTracker) List() ([]*resourceapi.ResourceClaim, error) {
+func (u *unsetResourceClaimTracker) List() ([]*resourceapi.ResourceClaim, error) {
 	return nil, fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) Get(namespace, claimName string) (*resourceapi.ResourceClaim, error) {
+func (u *unsetResourceClaimTracker) Get(namespace, claimName string) (*resourceapi.ResourceClaim, error) {
 	return nil, fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) ListAllAllocatedDevices() (sets.Set[structured.DeviceID], error) {
+func (u *unsetResourceClaimTracker) ListAllAllocatedDevices() (sets.Set[structured.DeviceID], error) {
 	return nil, fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) GatherAllocatedState() (*structured.AllocatedState, error) {
+func (u *unsetResourceClaimTracker) GatherAllocatedState() (*structured.AllocatedState, error) {
 	return nil, fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) SignalClaimPendingAllocation(claimUID types.UID, allocatedClaim *resourceapi.ResourceClaim) error {
+func (u *unsetResourceClaimTracker) SignalClaimPendingAllocation(claimUID types.UID, allocatedClaim *resourceapi.ResourceClaim) error {
 	return fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) ClaimHasPendingAllocation(claimUID types.UID) bool {
+func (u *unsetResourceClaimTracker) ClaimHasPendingAllocation(claimUID types.UID) bool {
 	klog.Errorf("lister not set in delegate")
 	return false
 }
 
-func (u unsetResourceClaimTracker) RemoveClaimPendingAllocation(claimUID types.UID) (deleted bool) {
+func (u *unsetResourceClaimTracker) RemoveClaimPendingAllocation(claimUID types.UID) (deleted bool) {
 	klog.Errorf("lister not set in delegate")
 	return false
 }
 
-func (u unsetResourceClaimTracker) AssumeClaimAfterAPICall(claim *resourceapi.ResourceClaim) error {
+func (u *unsetResourceClaimTracker) AssumeClaimAfterAPICall(claim *resourceapi.ResourceClaim) error {
 	return fmt.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceClaimTracker) AssumedClaimRestore(namespace, claimName string) {
+func (u *unsetResourceClaimTracker) AssumedClaimRestore(namespace, claimName string) {
 	klog.Errorf("lister not set in delegate")
 }
 
-func (u unsetResourceSliceLister) ListWithDeviceTaintRules() ([]*resourceapi.ResourceSlice, error) {
+func (u *unsetCSINodeLister) List() ([]*storagev1.CSINode, error) {
```

Comment:
Could you move the CSI methods after the DRA methods (after `unsetDeviceClassResolver.GetDeviceClass()`)?